### PR TITLE
Show X axis labels on charts

### PR DESCRIPTION
**Pull request by lyricnz**
Per #156

### CODE REVIEW

**Comment by lyricnz** [comment]
Example

Screen Shot 2021-09-27 at 8 05 30 am

**Comment by simon-anz** [comment]
Should labels be optional, and/or configurable?

**Comment by lyricnz** [comment]
@miguelmota Now that we have time-series labels, if we are charting a single asset, and we get less data (to the left) than requested, should we chart just what we have (with accurate labels), rather than resampling to what was requested?  This seems better (especially for "All Time" data).  I'll make that change - but feel free to revert the next commit.

File: pkg/timedata/timedata.go [hunk 1]
```diff
@@ -43,9 +41,25 @@ func CalculateTimeQuantum(data [][]float64) time.Duration {
...
 
-// Print out all the [timestamp,value] data provided
-func DebugLogPriceData(data [][]float64) {
+// BuildTimeSeriesLabels returns a list of short labels representing time values from the given [timestamp,value] data
+func BuildTimeSeriesLabels(data [][]float64) []string {
+	minTime := time.UnixMilli(int64(data[0][0]))
+	maxTime := time.UnixMilli(int64(data[len(data)-1][0]))
+	timeRange := maxTime.Sub(minTime)
+
+	var timeFormat string
```
Comment (miguelmota) [inline pkg/timedata/timedata.go:50]: This function looks really good.

The time format will depend on the user's region; for example in US the day is after the month. Maybe there's a library out there to convert to user's local format? 

The locale language to use can also be determined by host local environment variables. E.g:

DeepinScreenshot_select-area_20210927112336

For example, in [`pkg/humanize/humanize.go`](https://github.com/miguelmota/cointop/blob/master/pkg/humanize/humanize.go) we use `LC_NUMERIC` and `LC_MONETARY` for currency formatting. I think we can do something similar for date formatting.

Googling around found this [library](https://github.com/goodsign/monday) that is good for translating date language given locale. E.g.

DeepinScreenshot_select-area_20210927113213
DeepinScreenshot_select-area_20210927113224

I don't know anything about i18n it's up for discussion 😃

**Comment by miguelmota** [comment]
I think it's fine to keep the labels enabled since they make the chart a lot more useful. We can always make them optional or configurable if there's more demand from users

File: cointop/chart.go [hunk 2]
```diff
@@ -174,9 +174,10 @@ func (ct *Cointop) ChartPoints(symbol string, name string) error {
 
 	// Resample cachedata
 	timeQuantum := timedata.CalculateTimeQuantum(cacheData)
-	newStart := time.Unix(start, 0).Add(timeQuantum)
+	newStart := cacheData[0][0] // use the first data point
```
[the remark this answers is not in the document]
Comment (lyricnz) [reply cointop/chart.go:177]: Good point. Fixed.

**Pull request merged**